src/wallet/wallet.h

```diff
@@ -477,21 +477,21 @@ class CWalletTx : public CMerkleTx
 
 class CInputCoin {
 public:
-    CInputCoin()
-    {
-    }
     CInputCoin(const CWalletTx* walletTx, unsigned int i)
     {
-        if (walletTx != nullptr && i < walletTx->tx->vout.size())
+        if (walletTx == nullptr)
+            throw std::invalid_argument("walletTx should not be null");
+        if (i < walletTx->tx->vout.size())

```

Comment:
Can you invert the logic here `if (!P) throw` is a bit easier to read.

Might also be good to make it a `const CWalletTx&` so as to make sure that de-referencability is checked everywhere that this is called.